test/cases/ref.ts

```diff
@@ -0,0 +1,40 @@
+export var schema = 
+{
+  "title": "Referencing",
+  "type": "object",
+  "properties": {
+    "foo": {
+      "$ref": "test/resources/ReferencedType.json"
+    }
+  },
+  "required": ["foo"],
+  "additionalProperties": false
+}
+
+export var configurations = [
+  {
+    settings: {
+      declareReferenced: true
```

Comment:
The option already exists (I didn't add it), so there's other additional work to be done to remove it, which is outside the scope of this PR.  

And actually now that I read closer, this functionality is necessary if I want to keep my util classes:  

> Disallowed Merges
> Not all merges are allowed in TypeScript. Currently, classes can not merge with other classes or with variables. For information on mimicking class merging, see the Mixins in TypeScript section.